(chore): add changeset for version upgrade (#839)
hms-dbmi/viv: .changeset/chatty-ravens-collect.md

---
"@vivjs/constants": minor
"@vivjs/extensions": minor
"@vivjs/layers": minor
"@vivjs/loaders": minor
"@hms-dbmi/viv": minor
"@vivjs/types": minor
"@vivjs/viewers": minor
"@vivjs/views": minor
---

> (chore): upgrade deck.gl + luma.gl

Reviewer: One nit. I'd but more details here about specifically what changed (i.e., what versions, that this is breaking). With changesets you have the space to explain why the changes are involved. This is how we communicate to developers what something has changed.

For example I always appreciate esbuilds changelogs: https://github.com/evanw/esbuild/blob/main/CHANGELOG.md#0240

They let me know as the user what I can expect.

```suggestion
Upgrade deck.gl to XXXX and Luma.gl to XXXX

This upgrade is **breaking**. You will need to... 
```

Author: Great call, added some deatil.

---
"@vivjs/constants": minor
"@vivjs/extensions": minor
"@vivjs/layers": minor
"@vivjs/loaders": minor
"@hms-dbmi/viv": minor
"@vivjs/types": minor
"@vivjs/viewers": minor
"@vivjs/views": minor
---

> This release potentially includes backward-incompatible changes. To avoid automatically adopting such releases, ensure you are either pinning the exact version of @hms-dbmi/viv in your package.json file or using a version range syntax that only accepts patch updates, such as ~0.16.1. Refer to npm's semver documentation for details.

Reviewer: ```suggestion
**This release includes backward-incompatible changes**. To avoid automatically adopting such releases, ensure you are either pinning the exact version of `@hms-dbmi/viv` in your `package.json` file or using a version range syntax that only accepts patch updates, such as `~0.16.1`. Refer to npm's [semver documentation](https://docs.npmjs.com/cli/v6/using-npm/semver/) for details.
```

Just adding some highlighting. Also I removed "potentially", we know it does.

>     Migrate to deck.gl v9

Reviewer: ```suggestion
**Migrate to deck.gl v9**
```

> This update modifies our code internally to support deck.gl v9, dropping compatibility with deck.gl v8.

Reviewer: ```suggestion
This update modifies our code internally to support deck.gl v9, dropping compatibility with deck.gl v8. See the [release notes](https://deck.gl/docs/whats-new#deckgl-v90) to learn more.
```

> Impact:

Reviewer: ```suggestion
**Impact**:
```

> We have not made any public API changes from this except the version bump, and what these new versions require "internally" to your GL related code:

Reviewer: I mean if someone was using the ability to make custom extensions, that would be a breaking change, right? I just don't think we need to shy around the language about changes.

The TL;DR for me is that if you are relying deeply on Viv's layers and internals, this might be harder migration but hopefully Deck.gl 9 is on your radar and something you're working towards and understand the changes (and will welcome these).

If you are using the react components/data lodaers... much less so as you probably aren't working with deck directly.

Reviewer: Like deprecating WebGL1 is a totally a public breaking change, even if its something that is "internal".

Reviewer: I agree—I would be distinct about "breaking" and the relationship with public APIs (i.e., breaking changes doesn't mean we've changed interfaces). 

We should clearly state that this upgrade is "breaking" due to the underlying changes required to support deck.gl 9. While Viv's public APIs haven’t changed, these dependency updates can affect users and require migration effort.

> We haven’t changed Viv's public API, but the upgrade to deck.gl 9.0 is considered breaking due to changes in its dependencies, which may require updates to WebGL-related code (e.g., shaders, injections, constants). Here are potential issues users may face:

    - deprecating WebGL1
    - changing any public-facing GL-specific variables to use the new luma.gl backend-agnostic variables (such as `interpolation` on the `ImageLayer`)
    - shader injection-location name changes (i.e., `gl_fragColor` -> `fragColor`)

> Action:

Reviewer: ```suggestion
**Action**:
```

>     - You will need to upgrade to deck.gl `9.0.x` if it is specified separately or you use other `deck.gl` functionality indpendently of Viv as having two versions is not supported.  The above list of "internally" changing APIs was not exhaustive and only covers the Viv-related changes.  Please see https://deck.gl/docs/upgrade-guide#upgrading-to-v90 for more information.

Reviewer: ```suggestion
   - You will need to upgrade to deck.gl `9.0.x` if you use it directly as having multiple versions of deck.gl is not supported. The above list only includes changes internally to Viv and is not an exhaustive summary of all changes required for our migration. For full details on upgrading to deck.gl `9.0.x`, please refer to the [upgrade guide](https://deck.gl/docs/upgrade-guide#upgrading-to-v90).
```

>    - You will need to upgrade to deck.gl `9.0.x` if you use it directly as having multiple versions of deck.gl is not supported. The above list only includes changes internally to Viv and is not an exhaustive summary of all changes required for our migration. For full details on upgrading to deck.gl `9.0.x`, please refer to the [upgrade guide](https://deck.gl/docs/upgrade-guide#upgrading-to-v90).
>     - Pin a specific Viv version or semver range to prevent unintended updates.

Reviewer: ```suggestion
- You will need to upgrade to deck.gl `9.0.x` if you use it directly as having multiple versions of deck.gl is not supported. The above list only includes changes internally to Viv and is not an exhaustive summary of all changes required for our migration. For full details on upgrading to deck.gl `9.0.x`, please refer to the [upgrade guide](https://deck.gl/docs/upgrade-guide#upgrading-to-v90).
- Pin a specific Viv version or semver range to prevent unintended updates.
```
